### PR TITLE
Refactor accelerator profile state hooks to no longer manage its own selected

### PR DESCRIPTION
**Pull request by Gkrumbach07**
https://issues.redhat.com/browse/RHOAIENG-9368

## Description

refactors accelerator hook to now only show the current state of the resource's accelerator
moved all selected accelerator to the accelerator dropdown component

## How Has This Been Tested?

existing tests pass
and
1. create a accelerator from the admin panel
2. create workbench and make sure that the option shows up in the workbench under accelerators
3. check that save works
4. do this for model serving modal too
5. disable the accelerator from he admin page
6. edit the workbench and verify that the disabled option is still selectable
7. delete the accelerator
8. edit the workbench
9. verify that a use existing settings option is available

## Test Impact

there is existing test coverage

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/utilities/useAcceleratorProfileState.ts [hunk 1]
```diff
@@ -9,143 +9,155 @@ import {
...
-      count: 0,
-      useExisting: false,
-    });
+): AcceleratorProfileState => {
+  // const [acceleratorProfileState, setData] = useGenericObjectState<AcceleratorProfileState>({
+  // acceleratorProfiles: [],
+  // acceleratorProfile: undefined,
+  // count: 0,
+  // unknownProfileDetected: false,
+  // });
```
Comment (pnaik1) [inline frontend/src/utilities/useAcceleratorProfileState.ts:35]: Is it okay to remove these lines??

File: frontend/src/utilities/useAcceleratorProfileState.ts [hunk 2]
```diff
@@ -9,143 +9,155 @@ import {
...
-                    {
-                      key: 'nvidia.com/gpu',
-                      operator: TolerationOperator.EXISTS,
-                      effect: TolerationEffect.NO_SCHEDULE,
-                    },
-                  ],
+          if (migratedAcceleratorProfile) {
+            // setData('acceleratorProfile', migratedAcceleratorProfile);
+            // setData('count', Number(nvidiaAcceleratorRequests.count ?? 0));
+            // setData('unknownProfileDetected', false);
```
Comment (pnaik1) [inline frontend/src/utilities/useAcceleratorProfileState.ts:94]: same here

File: frontend/src/utilities/useAcceleratorProfileState.ts [hunk 3]
```diff
@@ -9,143 +9,155 @@ import {
...
+          // setData('acceleratorProfiles', [fakeAcceleratorProfile, ...acceleratorProfiles]);
+          // setData('count', Number(nvidiaAcceleratorRequests.count ?? 0));
+          // setData('unknownProfileDetected', false);
+          return {
+            acceleratorProfiles: [fakeAcceleratorProfile, ...acceleratorProfiles],
+            acceleratorProfile: fakeAcceleratorProfile,
+            count: Number(nvidiaAcceleratorRequests.count ?? 0),
+            unknownProfileDetected: false,
+          };
         }
```
Comment (pnaik1) [inline frontend/src/utilities/useAcceleratorProfileState.ts:133]: same here

File: frontend/src/utilities/useAcceleratorProfileState.ts (hunk 1 above)
Comment (Gkrumbach07) [reply frontend/src/utilities/useAcceleratorProfileState.ts:35]: yeah i can remove them

**Comment by pnaik1** [comment]
tested locally, working as expected
/lgtm

**Comment by jpuzz0** [comment]
Only reviewed code, which /lgtm

**Review by jpuzz0 — APPROVED** [review]

**Comment by Gkrumbach07** [comment]
/approve

I tested all cases
1. switching
2. removing
3. disabled profile
4. deleted profile

**Pull request merged**